pkg/gator/test/types.go

```diff
@@ -64,3 +72,29 @@ func (r *GatorResponses) Results() []*GatorResult {
 
 	return res
 }
+
+// YamlGatorResult is a GatorResult minues a level of indirection on
+// the ViolatingObject and with struct tags for yaml marshaling.
+type YamlGatorResult struct {
+	types.Result
+	ViolatingObject map[string]interface{} `yaml:"violatingObject"`
+	Trace           *string                `yaml:"trace,flow"`
+}
+
+// GetYamlFriendlyResults is a convenience func to remove a level of indirection between
+// unstructured.Unstructured and unstructured.Unstructured.Object when calling MarshalYaml.
+func GetYamlFriendlyResults(results []*GatorResult) []*YamlGatorResult {
+	var yamlResults []*YamlGatorResult
+
+	for _, r := range results {
+		yr := &YamlGatorResult{
+			Result:          r.Result,
+			ViolatingObject: r.ViolatingObject.DeepCopy().Object,
+			// Trace: fmt.Sprintf("%v", *r.Trace),
```

Comment:
This shouldn't be here 🙃 ; I will remove it when I respond to any review comments